src/oic/oic/provider.py

```diff
@@ -1228,21 +1224,20 @@ def do_client_registration(self, request, client_id, ignore=None):
                 ) = self._verify_sector_identifier(request)
             except InvalidSectorIdentifier as err:
                 return error_response("invalid_configuration_parameter", descr=str(err))
-        elif "redirect_uris" in request:
-            if len(request["redirect_uris"]) > 1:
-                # check that the hostnames are the same
-                host = ""
-                for url in request["redirect_uris"]:
-                    part = urlparse(url)
-                    _host = part.netloc.split(":")[0]
-                    if not host:
-                        host = _host
-                    else:
-                        if host != _host:
-                            return error_response(
-                                "invalid_configuration_parameter",
-                                descr="'sector_identifier_uri' must be registered",
-                            )
+        elif "redirect_uris" in request and len(request["redirect_uris"]) > 1:
+            # check that the hostnames are the same
+            host = ""
+            for url in request["redirect_uris"]:
+                part = urlparse(url)
+                _host = part.netloc.split(":")[0]
```

Comment:
I guess this would break for IPv6 literals as hostnames. But it is probably a non issue as no one issues certificates for them for TLS.

Reply:
I do agree with both points :)